lib/gds_metrics/middleware.rb

```diff
@@ -8,7 +8,20 @@ def initialize(app)
           use GDS::Metrics::Gzip
           use GDS::Metrics::Auth
 
-          use Prometheus::Client::Rack::Collector, registry: Proxy.new
+          rails_label_builder = proc do |env|
+            {
+              method: env['REQUEST_METHOD'].downcase,
+              host:   env['HTTP_HOST'].to_s,
+              path:   PathConverter.convert_rails_path_to_route(env['PATH_INFO'].to_s),
+            }
+          end
+
+          if defined?(Rails)
+            use Prometheus::Client::Rack::Collector, registry: Proxy.new, &rails_label_builder
+          else
+            use Prometheus::Client::Rack::Collector, registry: Proxy.new
```

Comment:
So does it mean we can do the same for non-rails apps?

Reply:
I think we could extend it to non rails apps if we wanted but would likely need to add custom logic for each framework we might be supporting. At the moment I don't think this is worth doing but may be needed in the future as more people use the client.

It doesn't appear that Rack itself offers a way to get a controller/view/route name given a path unfortunately